src/mjolnir.py

```diff
@@ -54,57 +48,14 @@ def _pebble_service(self) -> typing.Any:
         """
         return getattr(self._charm, "pebble_service", None)
 
-    def _update_peer_data(self, container: ops.model.Container) -> None:
-        """Update peer data if needed.
-
-        Args:
-            container: Synapse container.
-        """
-        # If there is no secret, we use peer relation data
-        # If there is secret, then we update the secret and add the secret id to peer data
-        peer_relation = self._charm.model.get_relation(PEER_RELATION_NAME)
-        if not peer_relation:
-            # there is no peer relation so nothing to be done
-            return
-
-        if JujuVersion.from_environ().has_secrets and not peer_relation.data[self._charm.app].get(
-            SECRET_ID
-        ):
-            # we can create secrets and the one that we need was not created yet
-            logger.debug("Adding secret")
-            admin_user = self.create_admin_user(container)
-            secret = self._charm.app.add_secret({SECRET_KEY: admin_user.access_token})
-            peer_relation.data[self._charm.app].update({SECRET_ID: secret.id})
-            return
-
-        if not JujuVersion.from_environ().has_secrets and not peer_relation.data[
-            self._charm.app
-        ].get(SECRET_KEY):
-            # we can't create secrets and peer data is empty
-            logger.debug("Updating peer relation data")
-            admin_user = self.create_admin_user(container)
-            peer_relation.data[self._charm.app].update({SECRET_KEY: admin_user.access_token})
-
-    def create_admin_user(self, container: ops.model.Container) -> User:
-        """Create an admin user.
-
-        Args:
-            container: Synapse container.
+    @property
+    def _admin_access_token(self) -> typing.Optional[str]:
```

Comment:
Note that if you change the `get_admin_access_token` method to throw an exception as suggested below, it's better not to use a property (as properties shouldn't thrown exceptions) or to add a try/catch block here

Reply:
Nice, changed.